assets/marketplace/main.js

```diff
@@ -0,0 +1,99 @@
+let vscode;
+try {
+    vscode = acquireVsCodeApi();
+} catch (error) {
+    
+}
+
+const app = new Vue({
+    el: '#app',
+    data: {
+        message: 'Hello Vue!',
+        searchInput: '',
+        selectedModule: {},
+        selectedTag: "",
+        moduleName: '',
+        modules: [],
+        endpoint: document.getElementById('app').getAttribute('data-endpoint'),
+        errorMessage: "",
+    },
+    created: async function () {
+        this.modules = await this.getModules();
+    },
+    methods: {
+        getModules: async function () {
+            return (await axios.get(`${this.endpoint}/api/v1/modules`)).data;
+        },
+        getModuleMetadata: async function (module) {
+            const data = (await axios.get(module.iotEdgeMetadataUrl)).data;
+            let repository = data.containerUri;
+            let defaultTag = data.tagsOrDigests[0];
+            let splitArr = data.containerUri.split(":");
+            if (splitArr.length > 1) {
+                const tag = splitArr.pop();
+                if (data.tagsOrDigests.includes(tag)) {
+                    repository = splitArr.join(":");
+                    defaultTag = tag;
+                }
+                

```

Comment:
remove empty line

Reply:
How about move the processing data logic to localserver, and localserver returns the processed data to front?

Reply:
As metadata is just a simple json file, so I fetch data directly and do some simple data process in fronted side.